lightning/src/routing/gossip.rs

```diff
@@ -1786,7 +1786,7 @@ where
 	}
 
 	fn test_node_counter_consistency(&self) {
-		#[cfg(debug_assertions)]
+		#[cfg(test)]
```

Comment:
I wonder if you could still do some of this by only verifying consistency on writes for non-test runs by gating the call in `read_only()` with cfg test?

Reply:
Ah right, so the write verification is the problem. And only calling `test_node_counter_consistency` on reads instead then?

Reply:
Agreed. Indeed, if the writes wouldn't have been the bottleneck, just asserting on writes would make more sense.